src/commands-history.lisp

```diff
@@ -0,0 +1,40 @@
+(defpackage #:reblocks/commands/history
+  (:use #:cl)
+  (:import-from #:reblocks/actions
+                #:make-action)
+  (:import-from #:reblocks/commands
+                #:add-command)
+  (:export #:browser-history-push-state
+           #:browser-history-back
+           #:browser-history-forward))
+
+(in-package :reblocks/commands/history)
+
+(defun browser-history-push-state (url &key (state "{}")
+                                         pop-action)
+  "Invoke history.pushState on the client.
+
+See: https://developer.mozilla.org/en-US/docs/Web/API/History_API"
+  (let ((action-code (when pop-action
+                       (etypecase pop-action
+                         (string pop-action)
+                         (function (reblocks/actions:make-action pop-action))))))
```

Comment:
In the docstring to browser-history-push-state I think we should mention that an action will be called with two arguments `location` and `state`.

Also, I'd prefer state argument of the `browser-history-push-state` to be a hash-table, but not sure if it will be deserialized into a hash-table propertly when action us called by the frontend.